### ISSUE TITLE
Distribution Agnostic Support

### ISSUE THREAD

**Issue by prologic** [tutumcloud/tutum-agent#14]
Are there any plans to support the Tutum Agent on any Distribution already configured to run teh Docker Engine correctly? This would get me using Tutum's CLI, API and Web UI again on various hosts I run Docker with and manage currently by hand via ssh and various tools.

**Comment by borjaburgos** [comment]
We support CentOS and Ubuntu now and more coming. What distribution do you use?

**Comment by prologic** [comment]
[CRUX](https://crux.nu) -- I maintain the [Docker Ports](https://crux.nu/portdb/?a=search&q=docker) for CRUX. This is typically what I deploy on on metal/hardware.

Other types of hosts I'm looking at are [RancherOS](http://rancher.com/rancher-os/)

**Comment by chgangaraju** [comment]
We need support for `Redhat (RHEL 7)`

**Comment by borjaburgos** [comment]
PRs for this are not only welcomed, but also very much appreciated.

**Comment by zedtux** [comment]
Would be nice to have Debian  : )

**Comment by prologic** [comment]
Bump; anyone working on this yet?

**Comment by fermayo** [comment]
We support the following distros:
* Ubuntu 13.04, 14.04, 15.04
* CentOS 7
* Fedora 21, 22
* Debian 8
* Red Hat Enterprise Linux 7

**Comment by prologic** [comment]
That's great! What about:

- [CRUX](https://crux.nu/)
- [Arch](https://www.archlinux.org/)
- [Gentoo](https://www.gentoo.org/)
- *and many others*?

**Comment by janeczku** [comment]
What about a dockerized version of the agent? Now that would be really distribution agnostic. IMHO the choice of distributions is far too narrow at the moment. Essentially all distros that are using sysVinit service manager can't be used and thats a bunch.

**Comment by prologic** [comment]
I agree; let's Dockerize this thing? Quite a few distros will also never move away from sysVinit or similar for both technical and philosophical reasons.

**Comment by borjaburgos** [comment]
http://github.com/tutumcloud/tutum-node

**Comment by prologic** [comment]
:+1: This now lets all of us (*no matter the host*) use and run a Tutum agent! :)

**Comment by Anthonyzou** [comment]
And a ARM build! For support with scaleway servers.

**Comment by janeczku** [comment]
@borjaburgos Tutum-node is really meant to be used with Vagrant on a local dev machine. It starts an additional docker service **inside** the container. This docker-in-docker approach is nothing i would use on a production server - just too buggy. 
I really want an image that contains just the tutum-agent which not launch a new docker service but will use the existing docker service on the host to launch all necessary containers (tutum/metrics, etc...) and deploy my service containers. This is really no problem. One would really just need to mount `/var/run/docker.sock` and tutum-agent will then be able to communicate with it.
